tools/scylla-sstable.cc

```diff
@@ -3249,9 +3258,26 @@ Validate the specified sstables:
         cache_tracker tracker;
         sstables::directory_semaphore dir_sem(1);
         abort_source abort;
-        sstables::sstables_manager sst_man("scylla_sstable", large_data_handler, dbcfg, feature_service, tracker,
-            memory::stats().total_memory(), dir_sem,
-            [host_id = locator::host_id::create_random_id()] { return host_id; }, abort);
+
+        sharded<sstables::storage_manager> sstm;
+        sstables::storage_manager::config stm_cfg;
+        stm_cfg.s3_clients_memory = std::clamp<size_t>(memory::stats().total_memory() * 0.01, 10 << 20, 100 << 20);
```

Comment:
scylla-sstable uses the standard memory allocator, *not* the seastar one. `memory::stats().total_memory()` is meaningless here.